### PR TITLE
[UI] Add CheckingPeriod field to channel create and update page

### PR DESCRIPTION
**Pull request by TheMilek**
image

### CODE REVIEW

File: src/Infrastructure/Form/Extension/ChannelTypeExtension.php [hunk 1]
```diff
@@ -27,6 +28,10 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'label' => 'sylius_price_history.form.channel.lowest_price_for_discounted_products_visible',
                 'required' => false,
             ])
+            ->add('lowestPriceForDiscountedProductsCheckingPeriod', IntegerType::class, [
+                'label' => 'sylius_price_history.form.channel.time_period_for_which_the_lowest_price_is_displayed',
+                'required' => false,
```
Comment (NoResponseMate) [inline src/Infrastructure/Form/Extension/ChannelTypeExtension.php:33]:
```suggestion
```

File: templates/Channel/Admin/Form/_additionalConfiguration.html.twig [hunk 2]
```diff
@@ -1,3 +1,4 @@
 <div class="ui attached segment">
     {{ form_row(form.lowestPriceForDiscountedProductsVisible) }}
+    {{ form_row(form.lowestPriceForDiscountedProductsCheckingPeriod) }}
```
Comment (NoResponseMate) [inline templates/Channel/Admin/Form/_additionalConfiguration.html.twig:3]: Would be nice to have the form widget/row overridden to make it a right-labeled input with `days` as the label, similar to how it's done in price inputs

File: tests/Behat/Element/Admin/Channel/DiscountedProductsCheckingPeriodInputElement.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public function getPeriod(): int
+    {
+        return (int) $this->getElement('discounted_products_checking_period')->getValue();
+    }
+
+    protected function getDefinedElements(): array
+    {
+        return [
+            'discounted_products_checking_period' => '#sylius_channel_lowestPriceForDiscountedProductsCheckingPeriod',
+        ];
```
Comment (NoResponseMate) [inline tests/Behat/Element/Admin/Channel/DiscountedProductsCheckingPeriodInputElement.php:34]: missing parent call

File: tests/Behat/Page/Admin/Channel/UpdatePage.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+use Sylius\Behat\Page\Admin\Channel\UpdatePage as BaseUpdatePage;
+
+class UpdatePage extends BaseUpdatePage implements UpdatePageInterface
+{
+    protected function getDefinedElements(): array
+    {
+        return [
+            'discounted_products_checking_period' => '#sylius_channel_lowestPriceForDiscountedProductsCheckingPeriod',
+        ];
```
Comment (NoResponseMate) [inline tests/Behat/Page/Admin/Channel/UpdatePage.php:24]: missing parent call

File: tests/Behat/Page/Admin/Channel/UpdatePageInterface.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\PriceHistoryPlugin\Behat\Page\Admin\Channel;
+
+use Sylius\Behat\Page\Admin\Channel\UpdatePageInterface as BaseUpdatePageInterface;
+
+interface UpdatePageInterface extends BaseUpdatePageInterface
+{
+}
```
Comment (NoResponseMate) [inline tests/Behat/Page/Admin/Channel/UpdatePageInterface.php:20]: Is this needed?

File: translations/messages.en.yaml [hunk 6]
```diff
@@ -5,6 +5,7 @@ sylius_price_history:
     form:
         channel:
             lowest_price_for_discounted_products_visible: 'Show the lowest price of discounted products prior to the current discount?'
+            time_period_for_which_the_lowest_price_is_displayed: 'Time period for which the lowest price is displayed:'
```
Comment (NoResponseMate) [inline translations/messages.en.yaml:8]:
```suggestion
            period_for_which_the_lowest_price_is_calculated: 'Period for which the lowest price is calculated'
```
or smth like that

File: tests/Behat/Resources/suites/ui/channel/managing_channels.yaml [hunk 7]
```diff
@@ -10,6 +10,8 @@ default:
                 - sylius.behat.context.transform.channel
                 - sylius.behat.context.transform.currency
                 - sylius.behat.context.transform.locale
+                - sylius.behat.context.transform.shared_storage
+                - sylius.behat.context.transform.country
```
Comment (NoResponseMate) [inline tests/Behat/Resources/suites/ui/channel/managing_channels.yaml:14]: order

File: templates/Channel/Admin/Form/_additionalConfiguration.html.twig [hunk 8]
```diff
@@ -1,3 +1,5 @@
+{% form_theme form '@SyliusPriceHistoryPlugin/Channel/Admin/Form/theme.html.twig' %}
 <div class="ui attached segment">
```
Comment (GSadee) [inline templates/Channel/Admin/Form/_additionalConfiguration.html.twig:2]:
```suggestion
{% form_theme form '@SyliusPriceHistoryPlugin/Channel/Admin/Form/theme.html.twig' %}

<div class="ui attached segment">
```

File: templates/Channel/Admin/Form/theme.html.twig [hunk 9]
```diff
@@ -0,0 +1,10 @@
+{% extends '@SyliusAdmin/Form/theme.html.twig' %}
+
+{% block _sylius_channel_lowestPriceForDiscountedProductsCheckingPeriod_widget %}
+    <div class="ui right labeled input">
+        {{ form_widget(form) }}
+        <div class="ui label">
+            {{ "sylius_price_history.ui.days"|trans }}
```
Comment (GSadee) [inline templates/Channel/Admin/Form/theme.html.twig:7]:
```suggestion
            {{ 'sylius_price_history.ui.days'|trans }}
```

File: tests/Behat/Resources/services/pages/channel.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+ (c) Paweł Jędrzejewski
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="sylius_price_history.behat.page.admin.channel.update" class="Tests\Sylius\PriceHistoryPlugin\Behat\Page\Admin\Channel\UpdatePage" parent="sylius.behat.page.admin.crud.update" public="true">
```
Comment (GSadee) [inline tests/Behat/Resources/services/pages/channel.xml:16]:
```suggestion
        <service id="Tests\Sylius\PriceHistoryPlugin\Behat\Page\Admin\Channel\UpdatePage" parent="sylius.behat.page.admin.crud.update" public="true">
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Kamil! :tada:

**Pull request merged**